.github/workflows/eslint-prettier-on-pr.yml

```diff
@@ -0,0 +1,31 @@
+# workflow to enforce style and linting on PR
+name: eslint-prettier
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
+jobs:
+  format-and-lint:
+    runs-on: ubuntu-latest
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
+      contents: write
+      
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      # Setup npm environment to run tests
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
+      - run: npx eslint-config-prettier src/*.ts src/**/*.ts
```

Comment:
Can we specify the target files in the eslint and prettier configs so we don't have pass them as CLI arguments both here and in `package.json`?

Reply:
Alternatively, can we just use scripts from `package.json` here since most of these are copies of what's already there?

Reply:
Yea I like the second idea using scripts from `package.json`